### PR TITLE
GraphQL - Better error messages formatting 

### PR DESCRIPTION
**Pull request by Premwoik**
This PR addresses [MIM-1569](https://erlangsolutions.atlassian.net/browse/MIM-1569) and solves wrongly formatted messages returned to the user from the webserver.

Proposed changes include:
* Unification of the error messages format.
* Introducing more information to authorization errors.
* Testing produced messages.
* Adding a catch that should handle all unexpected internal errors.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/94508) / small_tests / 866f87809d063a042cee909f5614629b063605d4
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3454/94508/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3454/94508/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-15_07.54.01/lib.mongooseim.logs/run.2021-12-15_07.54.02/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/94644) / small_tests / 385e621ddc651d10bcebf9b13b1893b558e52ebb
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3454/94644/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3454/94644/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-15_13.43.47/lib.mongooseim.logs/run.2021-12-15_13.43.47/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/94668) / small_tests / 5e16dec0f52dbf97fabbf6d9210288dee94c4c6a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3454/94668/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3454/94668/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-15_15.15.58/lib.mongooseim.logs/run.2021-12-15_15.15.58/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/94959) / small_tests / 5cf748d4f6d91cf3627abf1df9d520d66a89b56f
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3454/94959/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3454/94959/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-16_11.50.20/lib.mongooseim.logs/run.2021-12-16_11.50.21/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/95155) / small_tests / c2f50e4504da649ae4a5ffa543a95fca4e0cbb3d
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3454/95155/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3454/95155/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-16_16.25.50/lib.mongooseim.logs/run.2021-12-16_16.25.50/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/95202) / small_tests / b72180bc60038daec97adce977f15ec2d09db314
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3454/95202/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3454/95202/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-16_18.24.18/lib.mongooseim.logs/run.2021-12-16_18.24.19/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/95228) / small_tests / ec5bfd7d39dd66e1dcf20e29670aba7d338860b3
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3454/95228/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3454/95228/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-17_08.45.47/lib.mongooseim.logs/run.2021-12-17_08.45.48/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/95279) / small_tests / 6a73d1d30b179a575eb0d9f5569e66834f2efbc7
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3454/95279/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3454/95279/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-17_09.51.04/lib.mongooseim.logs/run.2021-12-17_09.51.04/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/95350) / small_tests / c0671d1c997b8864de7f24fe4d36567dc4994776
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3454/95350/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3454/95350/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-17_10.30.45/lib.mongooseim.logs/run.2021-12-17_10.30.45/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/95376) / small_tests / ae213925bce44084705e9357b9e318a8760f9579
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3454/95376/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3454/95376/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-17_10.34.18/lib.mongooseim.logs/run.2021-12-17_10.34.18/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/95403) / small_tests / 8584d4e0f7f3b30a365143d20e305e32ef40fa9f
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3454/95403/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3454/95403/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-17_10.44.04/lib.mongooseim.logs/run.2021-12-17_10.44.04/suite.log.html)

**Review by JanuszJakubiec — APPROVED** [review]

**Review by chrzaszcz — COMMENTED** [review]
The code looks good, error handling looks improved. I think some possible GraphQL errors are not fully tested - only formatting is checked. I guess it's just difficult to actually trigger all errors by faulty queries?

File: big_tests/tests/graphql_SUITE.erl [hunk 1]
```diff
@@ -98,7 +100,19 @@ wrong_creds_cannot_access_protected_types(Config) ->
...
-    assert_no_permissions(Status, Data).
+    assert_no_permissions(wrong_credentials, Status, Data).
+
+malformed_auth_header_error(Config) ->
+    EpName = ?config(schema_endpoint, Config),
+    Request =
+      #{port => get_port(EpName),
+        role => {graphql, atom_to_binary(EpName)},
+        method => <<"POST">>,
+        headers => [{<<"Authorization">>, <<"Basic YWRtaW46c2VjcmV">>}],
```
[the remark this answers is not in the document]
Comment (Premwoik) [reply big_tests/tests/graphql_SUITE.erl:111]: In this case, the value of the encoded credentials is malformed and cannot be decoded with base64. I added a comment about it

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/96110) / small_tests / 06bec49f962bb543cc43d32451f8c014b808aa5e
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3454/96110/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3454/96110/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-20_13.17.04/lib.mongooseim.logs/run.2021-12-20_13.17.04/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/96132) / small_tests / 88102c2b34a50bc5d28431772374b89d1ed7aa73
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3454/96132/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3454/96132/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-20_13.20.40/lib.mongooseim.logs/run.2021-12-20_13.20.41/suite.log.html)

**Comment by mongoose-im** [comment]
[small_tests_24](https://circleci.com/gh/esl/MongooseIM/96187) / small_tests / 6d971fe690402f094e57860dbc51da2ce5edd989
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/3454/96187/small_tests.24.0.5) / [small](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/3454/96187/small_tests.24.0.5/small/ct_run.mongooseim@localhost.2021-12-20_13.52.54/lib.mongooseim.logs/run.2021-12-20_13.52.54/suite.log.html)

**Pull request merged**